drivers/sensor/lsm6dsl/lsm6dsl.c

```diff
@@ -99,6 +165,153 @@ static int lsm6dsl_gyro_set_odr_raw(struct device *dev, u8_t odr)
 
+#ifdef LSM6DSL_ACCEL_ODR_RUNTIME
+static int lsm6dsl_accel_odr_set(struct device *dev, u16_t freq)
+{
+	int odr;
+
+	odr = lsm6dsl_freq_to_odr_val(freq);
+	if (odr < 0) {
+		return odr;
+	}
+
+	if (lsm6dsl_accel_set_odr_raw(dev, odr) < 0) {
+		SYS_LOG_DBG("failed to set accelerometer sampling rate");
+		return -EIO;
+	}
+
+	return 0;
+}
+#endif
+
+#ifdef LSM6DSL_ACCEL_FS_RUNTIME
+static int lsm6dsl_accel_range_set(struct device *dev, s32_t range)
+{
+	int fs;
+	struct lsm6dsl_data *data = dev->driver_data;
+
+	fs = lsm6dsl_accel_range_to_fs_val(range);
+	if (fs < 0) {
+		return fs;
+	}
+
+	if (lsm6dsl_accel_set_fs_raw(dev, fs) < 0) {
+		SYS_LOG_DBG("failed to set accelerometer full-scale");
+		return -EIO;
+	}
+
+	data->accel_sensitivity = (float)(lsm6dsl_accel_fs_sens[fs]
+						    * SENSI_GRAIN_XL);
+	return 0;
+}
+#endif
+
+static int lsm6dsl_accel_config(struct device *dev, enum sensor_channel chan,
+			    enum sensor_attribute attr,
+			    const struct sensor_value *val)
+{
+	switch (attr) {
+#ifdef LSM6DSL_ACCEL_FS_RUNTIME
+	case SENSOR_ATTR_FULL_SCALE:
+		return lsm6dsl_accel_range_set(dev, sensor_ms2_to_g(val));
+#endif
+#ifdef LSM6DSL_ACCEL_ODR_RUNTIME
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		return lsm6dsl_accel_odr_set(dev, val->val1);
+#endif
+	default:
+		SYS_LOG_DBG("Accel attribute not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+#ifdef LSM6DSL_GYRO_ODR_RUNTIME
+static int lsm6dsl_gyro_odr_set(struct device *dev, u16_t freq)
+{
+	int odr;
+
+	odr = lsm6dsl_freq_to_odr_val(freq);
+	if (odr < 0) {
+		return odr;
+	}
+
+	if (lsm6dsl_gyro_set_odr_raw(dev, odr) < 0) {
+		SYS_LOG_DBG("failed to set gyroscope sampling rate");
+		return -EIO;
+	}
+
+	return 0;
+}
+#endif
+
+#ifdef LSM6DSL_GYRO_FS_RUNTIME
+static int lsm6dsl_gyro_range_set(struct device *dev, s32_t range)
+{
+	int fs;
+	struct lsm6dsl_data *data = dev->driver_data;
+
+	fs = lsm6dsl_gyro_range_to_fs_val(range);
+	if (fs < 0) {
+		return fs;
+	}
+
+	if (lsm6dsl_gyro_set_fs_raw(dev, fs) < 0) {
+		SYS_LOG_DBG("failed to set gyroscope full-scale");
+		return -EIO;
+	}
+
+	data->gyro_sensitivity = (float)(lsm6dsl_gyro_fs_sens[fs]
+						    * SENSI_GRAIN_G);
+	return 0;
+}
+#endif
+
+static int lsm6dsl_gyro_config(struct device *dev, enum sensor_channel chan,
+			    enum sensor_attribute attr,
+			    const struct sensor_value *val)
+{
+	switch (attr) {
+#ifdef LSM6DSL_GYRO_FS_RUNTIME
+	case SENSOR_ATTR_FULL_SCALE:
+		return lsm6dsl_gyro_range_set(dev, sensor_ms2_to_g(val));
+#endif
+#ifdef LSM6DSL_GYRO_ODR_RUNTIME
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		return lsm6dsl_gyro_odr_set(dev, val->val1);
+#endif
+	default:
+		SYS_LOG_DBG("Gyro attribute not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int lsm6dsl_attr_set(struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
```

Comment:
Should we consider returning an error for single axis options, since it appears that all three axes must be the same?